nomad/constants.py

```diff
@@ -1,6 +1,12 @@
 # -*- coding: utf-8 -*-
 
 
+# PROJECTS
+# --
+
+METADATA_BASE_DIR = '.nomad'
```

Comment:
If I understand correctly then, this particular change (the addition of the `.nomad` folder) is not strictly required by this PR. It just happens to be bundled with it. We could have introduced this change earlier with the same justification, and we could introduce it later with the same justification.

As for the justification itself, it seems that we're faced with two "evils":

1. Enumerating through all containers at each `nomad` runs.
2. Polluting project folders (and force users to add yet another entry to all their `.gitignore`).

Unless there's a particular performance problem that hit us after a certain number of containers, I don't see how evil 1 is worse than evil 2.

There's an additional set of evils:

1. Lose container references when we move the project folder.
2. Lose container references when we do `git clean -fxd`.

But then again, it's hard to see which one is worse.